src/main/webapp/user-page.html

```diff
@@ -16,29 +16,67 @@
 
 <!DOCTYPE html>
 <html>
+  <head>
+    <title>User Page</title>
+    <meta charset="UTF-8">
+    <link rel="stylesheet" href="/css/main.css">
+    <link rel="stylesheet" href="/css/user-page.css">
+    <script src="/js/user-page-loader.js"></script>
+  </head>
 
-<head>
-  <title>User Page</title>
-  <meta charset="UTF-8">
-  <link rel="stylesheet" href="/css/main.css">
-  <link rel="stylesheet" href="/css/user-page.css">
-  <script src="/js/user-page-loader.js"></script>
-</head>
-
-<body onload="buildUI();">
-  <nav>
-    <ul id="navigation">
-      <li><a href="/">Home</a></li>
-    </ul>
-  </nav>
-  <h1 id="page-title">User Page</h1>
-
-  <form id="message-form" action="/messages" method="POST" class="hidden">
-    Enter a new message:
-    <br/>
-    <textarea name="text" id="message-input"></textarea>
-    <br/>
-    <input type="submit" value="Submit">
+  <body onload="buildUI();">
+    <nav>
+      <ul id="navigation">
+        <li><a href="/">Home</a></li>
+      </ul>
+    </nav>
+    <h1 id="page-title">User Page</h1>
+   
+    <jsp:directive.taglib uri="http://java.sun.com/jsp/jstl/core" prefix="c"%>
+    <jsp.directive.taglib uri="http://java.sun.com/jsp/jstl/functions" prefix="fn"%>
+<div class="userForm">
+  <h2>
+    Input what tea you drank today
+  </h2>
+
+  <form method="POST" action="/api/user-form-data">
+
+    <div>
+      <label for="greenTea">Green Tea</label>
+      <input type="number" name="greenTea" id="greenTea" size="40" value="${fn:escapeXml(tea.greenTea)}" class="form-control" />
```

Comment:
What's this escaping for?

Reply:
We are using JavaServer Page's ```escapeXml``` to counter XSS attacks. 

It's explained here : https://cloud.google.com/appengine/docs/standard/java/building-app/handling-form-data

``` 
This form sends the form data to the handler at the /create URL. 
You should use JSP's escapeXml functionality to counter cross-site scripting (XSS) attacks
 ```

Reply:
Thanks for the link.  So, I think there are two things that you are missing:

  1. In the example, the assumption is that the HTML code is being generated on the server
      from a template using JSP. That is, the example works because "${fn:...}" is being
      evaluated in some code that is processing the `*.jsp` file before returning it as an
      HTML file to the browser. In our code, there is no such preprocessing; the *.html file
      is being returned as a static file with no such dynamic, server-side evaluation (unless you've
      specially wired it up like that? If so, please point me to where in your code, you are
      causing the HTML to be processed using JSP). 

  2. The attack is possible and needs to be mitigated, because arbitrary user-supplied
      text was being injected into the HTML response, and this text could potentially
      include content like `<script>` tags that could execute arbitrary code. In this case,
      though, you are including numeric values that cannot possibly include `<script>` tags.

Reply:
If I am understanding this correctly for **2**; If we to call for the form to only accept number values, then we don't need ```escapeXml``` because it's not possibly to include ```<script>``` tags in that field. Preventing the XSS attack from happening in the user form. 

However, if I was to change the input type to text for **1**, then we would have to use the escapeXml functionality to counter the XSS attack. On top of that, I would need to implement something like server side rendering onto ```user-page.html``` so that we are processing the input (*I did not implement server side rendering / preprocessing*).

If it's the case where I don't have to worry about it in this case if we keep our input type as numbers, I think I can safely remove it.

Thank you for the insight and the comment!